### PR TITLE
fix: :bug: update getAppDataHash to rely on appDataInfoAtom

### PR DESCRIPTION
**Pull request by tukantje**
# Summary

Fixes the issue regarding quote endpoints incorrectly identifying themselves, causing security systems to assume we are being hit by a denial of service attack.

This change does two things;
* When `useAppData` runs, update a local variable to set the app data hash
* In `getAppDataHash` use this value if present

Why? Because not everywhere we use this value do we have the react available (i.e. api fetchers). Since we are not using jotai v2+, we cannot use the store api to extract this value properly. As such, this solution should serve us well.

# To Test

1. Open CoW Swap
2. Enter currency values to get a quote endpoint call made
3. Go to console, search for useAppData
4. appDataHash you see there, copy it somewhere
5. Go to network tab
6. Find a recent quote endpoint call
7. Go to request tab to see request body
8. appData field you see here should be identical to appDataHash you have saved

# Background
https://cowservices.slack.com/archives/C0361CDG8GP/p1682539030086899

### CODE REVIEW

File: src/custom/state/appData/atoms.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import { updateAppDataHash } from '@src/custom/constants/appDataHash'
```
Comment (shoom3301) [inline src/custom/state/appData/atoms.ts:1]:
```suggestion
import { updateAppDataHash } from 'constants/appDataHash'
```

**Review by anxolin — COMMENTED** [review]
Soft approve. 

Well, I think this will work but not necessarily I see it as the purest.

You are just updating this new `appDataHash`  with the info from the atom. I think the `appDataHash` was meant to return the "default appData" which shouldn't be used unless there's an error generating it (which shouldn't happen).

The real appData comes from a hook, which reads from the state. I think, it should flow down to the REST API call as part of the information.

Anyways, as mentioned, if you can test this and works, i won't offer opposition to this, since for now the assumption of "1 fixed app-data per user at a given time" holds true, this is why it will still work

**Comment by tukantje** [comment]
I agree fully. I think in its current iteration it is hard to address in other forms but if we update our state management library, we should have more options on this front anyways.

Our api call stuff are not hooks, so getting information from react world out is the main issue - which jotai v2 has a solution for at least.

This one is more of a "let's band aid the current issue" thing, sort of.

**Review by shoom3301 — APPROVED** [review]

**Review by elena-zh — APPROVED** [review]
Works as described!

**Comment by anxolin** [comment]
@tukantje u can merge if you want

**Pull request merged**